sdk/src/mock_bus.rs

```diff
@@ -1,57 +1,42 @@
 //! Mock message bus for tests
 use crate::match_topic::match_topic;
 use crate::message_bus::{MessageBounds, MessageBus, Subscription, SubscriptionBounds};
-use anyhow::Result;
+use anyhow::{bail, Result};
 use futures::future::BoxFuture;
-use std::collections::VecDeque;
 use std::sync::Arc;
-use tokio::sync::{Mutex, Notify};
+use tokio::sync::{mpsc, Mutex};
 use tracing::debug;
 
 pub struct PublishRecord<M: MessageBounds> {
     pub topic: String,
     pub message: Arc<M>,
 }
 
-#[derive(Clone)]
 pub struct MockSubscription<M> {
-    messages: Arc<Mutex<VecDeque<(String, Arc<M>)>>>,
-    notify: Arc<Notify>,
+    topic: String,
+    receiver: mpsc::UnboundedReceiver<(String, Arc<M>)>,
```

Comment:
This is a convention I tend to follow with tokio channels: qualify the names with the packages they came from (e.g. call this type `mpsc::UnboundedReceiver` rather than `UnboundedReceiver`). That makes it easier to tell what the semantics of the channel are at a glance (you can see that a sender is `mpsc::Sender` vs `watch::Sender`), especially in crates which use more than one type of channel.